### PR TITLE
Bump Lightning and PennyLane minimum versions (v0.37)

### PR DESCRIPTION
**Pull request by rauletorresc**

### CODE REVIEW

File: .github/workflows/build-wheel-linux-x86_64.yaml [hunk 1]
```diff
@@ -316,7 +316,7 @@ jobs:
               -Dpybind11_DIR=$(python${{ matrix.python_version }} -c "import pybind11; print(pybind11.get_cmake_dir())") \
               -DENABLE_LIGHTNING_KOKKOS=ON \
               -DENABLE_LAPACK=OFF \
-              -DLIGHTNING_GIT_TAG=476cac2e39f79c122b8751a77aad8139b39296db \
+              -DLIGHTNING_GIT_TAG=latest_release \
```
Comment (vincentmr) [inline .github/workflows/build-wheel-linux-x86_64.yaml:319]: This is still v0.36, is that intended?

Comment (rauletorresc) [reply .github/workflows/build-wheel-linux-x86_64.yaml:319]: @erick-xanadu some thoughts on this?

Comment (dime10) [reply .github/workflows/build-wheel-linux-x86_64.yaml:319]: This PR will be merged after the lightning release

**Comment by erick-xanadu** [comment]
@rauletorresc you should also install pennylane before installing Catalyst

**Comment by erick-xanadu** [comment]
```
   - name: Install Catalyst
      run: |
        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
```

Here, I think you need to use --no-deps to install catalyst without dependencies and then you can install pennylane from RC here. Or the other way around. Install pennylane RC before hitting this line.

**Review by jay-selby — APPROVED** [review]

**Review by dime10 — COMMENTED** [review]
Thanks @rauletorresc, looks good to me!

**Pull request merged**